seatunnel-connectors-v2/connector-http/connector-http-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/http/client/HttpClientProvider.java

```diff
@@ -115,14 +120,42 @@ public HttpResponse execute(
             String method,
             Map<String, String> headers,
             Map<String, String> params,
-            String body)
+            Map<String, Object> body,
+            Map<String, Object> pageParams,
+            boolean keepParamsAsForm)
             throws Exception {
         // convert method option to uppercase
         method = method.toUpperCase(Locale.ROOT);
+        // Keep the original post  logic
+        if (HttpPost.METHOD_NAME.equals(method) && keepParamsAsForm) {
+            URI uri = URI.create(url);
+            Map<String, Object> paramsMap = new HashMap<>();
+            if (MapUtils.isNotEmpty(params)) {
+                paramsMap.putAll(params);
+            }
+            if (MapUtils.isNotEmpty(pageParams)) {
+                paramsMap.putAll(params);
+            }
```

Comment:
when enable `keepParamsAsForm`, why not put params into request body? Then add header `application/x-www-form-urlencoded`. So that we can handle it with normally logic.